### ISSUE TITLE
PatternSyntaxExceptionorg.mozilla.fenix.components.Core$engine$2

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#3545]
```
java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 4:
(?<major>[0-9]+)\.(?<minor>[0-9]+)(?:\.(?<patch>[0-9]+))?(?<metadata>[^0-9].*)?
    ^
    at java.util.regex.Pattern.compileImpl(Pattern.java)
    at java.util.regex.Pattern.compile(Pattern.java:411)
    at java.util.regex.Pattern.<init>(Pattern.java:394)
    at java.util.regex.Pattern.compile(Pattern.java:381)
    at kotlin.text.Regex.<init>(Regex.kt:3)
    at mozilla.components.concept.engine.utils.EngineVersion.parse(EngineVersion.kt:1)
    at mozilla.components.browser.engine.gecko.GeckoEngine.<init>(GeckoEngine.kt:9)
    at org.mozilla.fenix.components.Core$engine$2.invoke(Core.kt:10)
    at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:6)
    at org.mozilla.fenix.components.Core.getEngine(Core.kt)
    at org.mozilla.fenix.components.Core$sessionManager$2.invoke(Core.kt:1)
    at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:6)
    at org.mozilla.fenix.components.Core.getSessionManager(Core.kt)
    at org.mozilla.fenix.FenixApplication.onTrimMemory(FenixApplication.kt:3)
    at android.app.ActivityThread.handleTrimMemory(ActivityThread.java:4573)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1663)
    at android.os.Handler.dispatchMessage(Handler.java:111)
    at android.os.Looper.loop(Looper.java:194)
    at android.app.ActivityThread.main(ActivityThread.java:5666)
    at java.lang.reflect.Method.invoke(Method.java)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:959)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:754)
```

**Comment by pocmo** [comment]
https://sentry.prod.mozaws.net/operations/fenix-nightly/issues/5986156/

**Comment by pocmo** [comment]
GeckoView: 68.0-20190612114833

**Comment by pocmo** [comment]
It looks like this could be Android OS version related. In the crash reports there are only 5.0.x and 5.1.x versions so far.

**Comment by pocmo** [comment]
We added this in 1.0.0 and should fix that in 2.0.0 asap.

**Comment by pocmo** [comment]
I can reproduce this with an Android 5 emulator.
